### ISSUE TITLE
host_topology: usage of strncat might overflow buffer

### ISSUE THREAD

**Issue by devreal** [dash-project/dash#57]
GCC issues the following warning for the function `dart__base__host_topology__module_locations`:

```
In function ‘strncat’,
    inlined from ‘dart__base__host_topology__module_locations’ at /home/joseph/src/dash/dash/dart-impl/base/src/internal/host_topology.c:103:13:
/usr/include/x86_64-linux-gnu/bits/string3.h:156:10: warning: call to __builtin___strncat_chk might overflow destination buffer
   return __builtin___strncat_chk (__dest, __src, __len, __bos (__dest));
```

The way strncat is used here is dangerous since the size parameter does not describe the size of the destination buffer. The man page reads:

> The strncat() function is similar, except that
>        *  it will use at most n bytes from src; [...]

A safe way would be to use snprintf or the like to control the amount of data written and thus to avoid memory corruption.

**Comment by fuchsto** [comment]
Hm? I don't get it. Actually it's the source length I want to guard as the destination buffer size is known.
Will have a look at this.

**Comment by fuchsto** [comment]
Ah, sure, I had it the other way around. Will be merged from branch bug-57-host-strncat.
Thanks! (Also, why did my gcc not report this? Which version do you fancy?)

**Comment by fuchsto** [comment]
Argh, sorry. Yes, as I'm _appending_, restricting the size of the snippets to append does not help much.
Thank you for reporting! Will be merged from branch `bug-57-host-strncat`

**Comment by fuchsto** [comment]
Fixed in PR #78
